src/ansys/fluent/core/transcript.py

```diff
@@ -0,0 +1,71 @@
+import itertools
+import threading
+from typing import Callable, Optional
+
+from ansys.fluent.core.services.transcript import TranscriptService
+
+lock = threading.Lock()
+
+
+class Transcript:
+    """Encapsulates a Fluent Transcript streaming service."""
+
+    def __init__(self, channel, metadata):
+        self._channel = channel
+        self._metadata = metadata
+        self.transcript_service = TranscriptService(self._channel, self._metadata)
+        self._transcript_thread: Optional[threading.Thread] = None
+        self._transcript_callbacks = {}
+        self._id_count = itertools.count()
+        self._transcript_callback_id = next(self._id_count)
+
+    def add_transcript_callback(self, callback_fn: Callable, keep_new_lines=False):
+        """Initiates a fluent transcript streaming depending on the
+        callback_fn.
+
+        For eg.: add_transcript_callback(print) prints the transcript on
+        the interpreter screen.
+        """
+        with lock:
+            self._transcript_callbacks[self._transcript_callback_id] = (
+                callback_fn,
+                keep_new_lines,
+            )
+            returned_callback_id = self._transcript_callback_id
+            self._transcript_callback_id = next(self._id_count)
+        if len(self._transcript_callbacks) == 1:
```

Comment:
The size of `self._transcript_callbacks` can change after the lock is released at the end of the with block, which would affect this logic.